### PR TITLE
Update Gatekeeper CRD descriptions

### PR DESCRIPTION
**Pull request by dhaiducek**
This represents a major update to the descriptions and reorganization of the `*_types.go` file, placing dependent structs higher in the file (and closer to structs that require them where relevant). I couldn't find where `status` was even used, so I've gutted it. I can revert that if there are concerns, but it seemed relatively safe since the operator should be the only thing populating it. I also made an effort to link to the upstream Gatekeeper docs wherever possible, which I think will go a long way toward enabling serviceability and helpfulness of the operator configuration.

ref: https://issues.redhat.com/browse/ACM-8992

### CODE REVIEW

**Review by JustinKuli — DISMISSED** [review]
/hold if anyone else wants to check, but it looks good to me!

**Review by yiraeChristineKim — APPROVED** [review]
Great!!

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -25,7 +25,10 @@ spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
     owned:
-    - description: Gatekeeper is the Schema for the gatekeepers API
+    - description: Gatekeeper is the Schema for the gatekeepers API. Gatekeeper contains
+        configurations for the Gatekeeper operator, which deploys Open Policy Agent
+        Gatekeeper, a policy engine that enforces policies through admission controller
+        webhooks. (See https://github.com/open-policy-agent/gatekeeper)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:31]:
```suggestion
    - description: Gatekeeper is the schema for the gatekeepers API. Gatekeeper contains
        configurations for the Gatekeeper operator, which deploys Open Policy Agent
        Gatekeeper, a policy engine that enforces policies through admission controller
        webhooks. See https://github.com/open-policy-agent/gatekeeper.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -34,37 +37,48 @@ spec:
         name: gatekeeper-deployment
         version: v1
       specDescriptors:
-      - displayName: Affinity
+      - description: Affinity is a node affinity specification to add to the Gatekeeper
+          deployment Pods. (See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:41]:
```suggestion
      - description: Affinity is a node affinity specification to add to the Gatekeeper
          deployment Pods. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -34,37 +37,48 @@ spec:
...
+        displayName: Affinity
         path: affinity
-      - displayName: Audit Configuration
+      - description: Audit specifies the configuration for the Gatekeeper auditing
+          function. (See https://open-policy-agent.github.io/gatekeeper/website/docs/audit)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:45]:
```suggestion
      - description: Audit specifies the configuration for the Gatekeeper auditing
          function. See https://open-policy-agent.github.io/gatekeeper/website/docs/audit.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -34,37 +37,48 @@ spec:
...
-      - displayName: Image Configuration
+      - description: Image specifies the configuration for handling Gatekeeper deployment
+          images.
+        displayName: Image Configuration
         path: image
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:hidden
-      - displayName: Mutating Webhook
+      - description: MutatingWebhook specifies whether the Gatekeeper mutating admission
+          webhook is enabled. Defaults to Enabled. (See https://open-policy-agent.github.io/gatekeeper/website/docs/mutation)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:59]:
```suggestion
          webhook is enabled. The default value is Enabled. See https://open-policy-agent.github.io/gatekeeper/website/docs/mutation.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -34,37 +37,48 @@ spec:
...
+        displayName: Mutating Webhook
         path: mutatingWebhook
-      - displayName: Node Selector
+      - description: NodeSelector is a map of node selectors to add to the Gatekeeper
+          deployment Pods. (See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:63]:
```suggestion
      - description: NodeSelector is a map of node selectors to add to the Gatekeeper
          deployment Pods. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -34,37 +37,48 @@ spec:
...
+        displayName: Node Selector
         path: nodeSelector
-      - displayName: Pod Annotations
+      - description: PodAnnotations is a map of additional annotations to be added
+          to the Gatekeeper deployment Pods.
+        displayName: Pod Annotations
         path: podAnnotations
-      - displayName: Tolerations
+      - description: Tolerations is an array of Tolerations to add to the Gatekeeper
+          deployment Pods. (See https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:71]:
```suggestion
      - description: Tolerations is an array of tolerations to add to the Gatekeeper
          deployment Pods. See https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/.
```

File: bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml [hunk 7]
```diff
@@ -34,37 +37,48 @@ spec:
...
+        displayName: Tolerations
         path: tolerations
-      - displayName: Validating Webhook
+      - description: ValidatingWebhook specifies whether the Gatekeeper validating
+          admission webhook is enabled. Defaults to Enabled.
```
Comment (dockerymick) [inline bundle/manifests/gatekeeper-operator.clusterserviceversion.yaml:75]:
```suggestion
      - description: ValidatingWebhook specifies whether the Gatekeeper validating
          admission webhook is enabled. The default value is Enabled.
```

File: config/manifests/bases/gatekeeper-operator.clusterserviceversion.yaml [hunk 8]
```diff
@@ -13,7 +13,10 @@ spec:
   apiservicedefinitions: {}
   customresourcedefinitions:
     owned:
-    - description: Gatekeeper is the Schema for the gatekeepers API
+    - description: Gatekeeper is the Schema for the gatekeepers API. Gatekeeper contains
```
Comment (dockerymick) [inline config/manifests/bases/gatekeeper-operator.clusterserviceversion.yaml:16]: Seems like these are similar descriptions from the `operator.gatekeeper.sh_gatekeeperrs.yaml. 

If that is accurate, I suggestion similar feedback

**Review by dockerymick — CHANGES_REQUESTED** [review]

File: api/v1alpha1/gatekeeper_types.go [hunk 9]
```diff
@@ -43,205 +43,289 @@ func (m Mode) ToBoolString() string {
...
-	Webhook *WebhookConfig `json:"webhook,omitempty"`
+const (
+	AuditFromCacheEnabled   AuditFromCacheMode = "Enabled"
+	AuditFromCacheDisabled  AuditFromCacheMode = "Disabled"
+	AuditFromCacheAutomatic AuditFromCacheMode = "Automatic"
+)
 
-	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Node Selector"
+type AuditConfig struct {
+	// AuditInterval configures how often audit is run on the cluster. The default value is 60s.
```
Comment (mprahl) [inline api/v1alpha1/gatekeeper_types.go:69]:
```suggestion
	// AuditInterval configures how often an audit is run on the cluster. The default value is 60s.
```

File: api/v1alpha1/gatekeeper_types.go [hunk 10]
```diff
@@ -43,205 +43,289 @@ func (m Mode) ToBoolString() string {
...
 	Operations []OperationType `json:"operations,omitempty"`
+
+	// DisabledBuiltins is a list of specific OPA built-in functions to disable. By default, http.send
+	// is disabled.
+	// See https://open-policy-agent.github.io/gatekeeper/website/docs/customize-startup#disable-opa-built-in-functions.
+	//
 	// +optional
 	DisabledBuiltins []string `json:"disabledBuiltins,omitempty"`
+
+	// LogMutations enables logging of mutation events and errors. The default value is Disabled.
```
Comment (mprahl) [inline api/v1alpha1/gatekeeper_types.go:180]:
```suggestion
	// LogMutations enables the logging of mutation events and errors. The default value is Disabled.
```

File: api/v1alpha1/gatekeeper_types.go [hunk 11]
```diff
@@ -43,205 +43,289 @@ func (m Mode) ToBoolString() string {
...
 
-const (
-	LogLevelDEBUG   LogLevelMode = "DEBUG"
-	LogLevelInfo    LogLevelMode = "INFO"
-	LogLevelWarning LogLevelMode = "WARNING"
-	LogLevelError   LogLevelMode = "ERROR"
-)
+type ConfigConfig struct {
+	// Matches is a list of exempt namespaces for specified processes. If specified, the Matches will
+	// be appended to the existing Config spec.match.
```
Comment (mprahl) [inline api/v1alpha1/gatekeeper_types.go:227]: It doesn't actually append it. It just appends to the default matches.
```suggestion
	// be set on the existing Config spec.match.
```

**Review by mprahl — APPROVED** [review]
Nice work!

File: config/crd/bases/operator.gatekeeper.sh_gatekeepers.yaml [hunk 12]
```diff
@@ -806,50 +813,81 @@ spec:
...
                     format: int64
                     minimum: 0
                     type: integer
                   auditEventsInvolvedNamespace:
+                    description: |-
+                      AuditEventsInvolvedNamespace controls which namespace audit events will be created in. When set
+                      to Enabled, audit events will be created in the namespace of the object violating the
+                      constraint. Setting to Disabled will cause all audit events to be created in the Gatekeeper
+                      namespace. The default value is Disabled.
+                      See https://open-policy-agent.github.io/gatekeeper/website/docs/customize-startup/#alpha-emit-admission-and-audit-events.
```
Comment (dockerymick) [inline config/crd/bases/operator.gatekeeper.sh_gatekeepers.yaml:834]: made a similar suggestion earlier.

**Review by dockerymick — CHANGES_REQUESTED** [review]
I realize that i didn't review another YAML file

**Review by mprahl — APPROVED** [review]

**Comment by dhaiducek** [comment]
Thanks for all the reviews, everyone! 😄

**Pull request merged**